unit_tests/fake/fake_error_manager.cpp

```diff
@@ -31,7 +34,7 @@ _er_log_debug (const char *file_name, const int line_no, const char *fmt, ...)
 int
 er_errid (void)
 {
-  return 0;
+  return error_id;
```

Comment:
- Set the error in er_set_with_oserror too and reset it in er_clear.
- Name error_id as g_error_id